pci/src/vfio_user.rs

```diff
@@ -391,6 +391,47 @@ impl Vfio for VfioUserClientWrapper {
             )
             .map_err(VfioError::VfioUser)
     }
+
+    fn set_irq_resample_fd(
+        &self,
+        irq_index: u32,
+        event_rfds: Vec<&EventFd>,
+    ) -> Result<(), VfioError> {
+        info!(
+            "Setting IRQ resample fd {:x} number of fds = {:?}",
+            irq_index,
+            event_rfds.len()
+        );
+        let fds: Vec<i32> = event_rfds.iter().map(|e| e.as_raw_fd()).collect();
+
+        // Batch into blocks of 16 fds as sendmsg() has a size limit
+        let mut sent_fds = 0;
+        let num_fds = event_rfds.len() as u32;
+        while sent_fds < num_fds {
+            let remaining_fds = num_fds - sent_fds;
+            let count = if remaining_fds > 16 {
+                16
+            } else {
+                remaining_fds
+            };
+
+            self.client
+                .lock()
+                .unwrap()
+                .set_irqs(
+                    irq_index,
+                    VFIO_IRQ_SET_DATA_EVENTFD | VFIO_IRQ_SET_ACTION_UNMASK,
+                    sent_fds,
+                    count,
+                    &fds[sent_fds as usize..(sent_fds + count) as usize],
+                )
+                .map_err(VfioError::VfioUser)?;
+
+            sent_fds += count;
+        }
+
```

Comment:
This part can be simplified a lot. There will always only be one resamplefd, so you don't need the loop here.

Reply:
> This part can be simplified a lot. There will always only be one resamplefd, so you don't need the loop here.

@rbradford I have removed the loop in the latest patch.